### PR TITLE
feat: Add download limit

### LINKED ISSUE #43566: Move the files_downloadlimit setting to new location
Subtask to #6841

### PR DESCRIPTION
**Pull request by Pytal**
- Resolve https://github.com/nextcloud/server/issues/43566

## TODO

- [ ] ...

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/files_sharing/src/services/DownloadLimitService.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	const response = await axios.put(generateOcsUrl('/apps/files_downloadlimit/api/v1/{token}/limit', { token }), {
+		limit,
+	})
+	return response.data.ocs.data
+}
+
+export const deleteDownloadLimit = async (token: string) => {
+	const response = await axios.delete(generateOcsUrl('/apps/files_downloadlimit/api/v1/{token}/limit', { token }))
+	return response.data.ocs.data
+}
```
Comment (skjnldsv) [inline apps/files_sharing/src/services/DownloadLimitService.ts:49]: Scoped app APIs calls into another app is something we should avoid :(

Comment (Pytal) [reply apps/files_sharing/src/services/DownloadLimitService.ts:49]: What alternative would you suggest? Merging the files_downloadlimit into files_sharing?

Comment (skjnldsv) [reply apps/files_sharing/src/services/DownloadLimitService.ts:49]: > Merging the files_downloadlimit app into files_sharing?

Seeing how closely it starts to be to the core functionalities, could actually be a simpler option yeah :/
More importantly, we already have an API for this, so if the app suddenly step away from this API, what is going to happen of it?

For me there is only two acceptable options
1. (preferably), the current API gets improved so the designers can also work their magic
2. another API gets implemented and we deprecate the previous one (can quickly be messy, not fond of that one)
3. we merge the files_downloadlimit into files_sharing (and we deprecate the API?)

Comment (Pytal) [reply apps/files_sharing/src/services/DownloadLimitService.ts:49]: Going with 1 as discussed

**Pull request closed without merging**